_search-plugins/ltr/advanced-functionality.md

```diff
@@ -0,0 +1,573 @@
+feature store, but wiktionary would be backed by another. There\'s
+nothing that would be shared between the two.
+
+Should your OpenSearch cluster back multiple properties, you can use
+all the capabilities of this guide on named feature stores, simply by:
+
+    PUT _ltr/wikipedia
+
+Then the same API in this guide applies to this feature store, for
+example to create a feature set:
+
+    POST _ltr/wikipedia/_featureset/attempt_1
+    {
+       "featureset": {
+            "features": [
+                {
+                    "name": "title_query",
+                    "params": [
+                        "keywords"
+                    ],
+                    "template_language": "mustache",
+                    "template": {
+                        "match": {
+                            "title": "{{keywords}}"
+                        }
+                    }
+                }
+            ]
+       }
+    }
+
+And of course you can delete a featureset:
+
+    DELETE _ltr/wikipedia/_featureset/attempt_1
+
+You can use featuresets of specific feature stores by using the `store`
+parameter in the `sltr` part of your query when logging features:
+
+    "sltr": {
+        "_name": "logged_featureset",
+        "featureset": "attempt_1",
+        "store": "wikipedia",
+        "params": {
+            "keywords": "star"
+        }
+    }
+
+In case no `store` is specified the default store will be used for
+looking up the featureset.
+
+## Model Caching
+
+The plugin uses an internal cache for compiled models.
+
+Clear the cache for a feature store to force models to be recompiled:
+
+    POST /_ltr/_clearcache
+
+Get cluster wide cache statistics for this store:
+
+    GET /_ltr/_cachestats
+
+Characteristics of the internal cache can be controlled with these node
+settings:
+
+    # limit cache usage to 12 megabytes (defaults to 10mb or max_heap/10 if lower)
+    ltr.caches.max_mem: 12mb
+    # Evict cache entries 10 minutes after insertion (defaults to 1hour, set to 0 to disable)
+    ltr.caches.expire_after_write: 10m
+    # Evict cache entries 10 minutes after access (defaults to 1hour, set to 0 to disable)
+    ltr.caches.expire_after_read: 10m
+
+## Extra Logging
+
+As described in [logging features]({{site.url}}{{site.baseurl}}/search-plugins/ltr/logging-features/), it is
+possible to use the logging extension to return the feature values with
+each document. For native scripts, it is also possible to return extra
+arbitrary information with the logged features.
+
+For native scripts, the parameter `extra_logging` is injected into the
+script parameters. The parameter value is a
+[Supplier](https://docs.oracle.com/javase/8/docs/api/java/util/function/Supplier.html)
+\<[Map](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html)\>,
+which provides a non-null `Map<String,Object>` **only** during the
+logging fetch phase. Any values added to this Map will be returned with
+the logged features:
+
+    @Override
```

Comment:
@sstults can you look at this?   @vagimeli  can you give me a bit more context of what you mean by validation?

Reply:
The code snipped is Java code and is only intended for explanation purposes. The confusion probably arises from the highlighting as this code snippet is marked as json instead of java. I will correct this. The snippet contains proper code.